hadoop-hdfs-project/hadoop-hdfs/src/main/java/io/hops/transaction/lock/BatchedINodeLock.java

```diff
@@ -17,37 +17,41 @@
 
 import io.hops.common.INodeUtil;
 import io.hops.exception.StorageException;
+import io.hops.exception.TransactionContextException;
 import io.hops.metadata.hdfs.entity.INodeIdentifier;
+import io.hops.transaction.EntityManager;
 import org.apache.hadoop.hdfs.server.namenode.INode;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 public class BatchedINodeLock extends BaseINodeLock {
 
   private final List<INodeIdentifier> inodeIdentifiers;
+  private final TransactionLockTypes.INodeLockType lockType;
   private long[] inodeIds;
 
-  public BatchedINodeLock(List<INodeIdentifier> inodeIdentifiers) {
+  public BatchedINodeLock(List<INodeIdentifier> inodeIdentifiers,
+                          TransactionLockTypes.INodeLockType lockType) {
     this.inodeIdentifiers = inodeIdentifiers;
-    inodeIds = new long[inodeIdentifiers.size()];
+    this.inodeIds = new long[inodeIdentifiers.size()];
+    this.lockType = lockType;
   }
 
   @Override
   protected void acquire(TransactionLocks locks) throws IOException {
     if (inodeIdentifiers != null && !inodeIdentifiers.isEmpty()) {
-      String[] names = new String[inodeIdentifiers.size()];
-      long[] parentIds = new long[inodeIdentifiers.size()];
-      long[] partitionIds = new long[inodeIdentifiers.size()];
-      for (int i = 0; i < inodeIdentifiers.size(); i++) {
-        INodeIdentifier inodeIdentifier = inodeIdentifiers.get(i);
-        names[i] = inodeIdentifier.getName();
-        parentIds[i] = inodeIdentifier.getPid();
-        partitionIds[i] = inodeIdentifier.getPartitionId();
-        inodeIds[i] = inodeIdentifier.getInodeId();
+
+      List<INode> inodes;
+      if(lockType == TransactionLockTypes.INodeLockType.READ_COMMITTED){
+        // batch read does not guarantee lock order
+        inodes = batchRead();
+      } else{
+        inodes = orderedReadWithLock();
```

Comment:
I wonder if we should not create a new lock object with a different name for this. The batched name let us believe that it will be a batch operation (fast) while it is a sequential operation (slow) the difference of performance can be quite important and I think it would be marked more clearly by having one called batched and the other called something else.

Reply:
I checked the remaining of the code and it is even worse than that, `lockType` is never `READ_COMMITTED`. This raises several questions:
1. Why not remove the batched part, use only the new code and rename the class so that it is clear that it is not batched.
2. What is the impact on the system performances?
3. How come we thought we could do without taking a write lock before? Do we really always need a write lock?

Reply:
Previously, batch read (without locking) was only used while namenode was in safemode. But now I see it is being called outside safemode.

Reply:
I agree the class should be renamed or a new class should be created for this purpose

Reply:
Performance impact will be significant. But it may not matter for background operations. I will look into synchronous operations.

Reply:
For performance we can increase the number of slicer threads. Currently it is set to 20 which may not be sufficient

Reply:
For the threads number, I remember playing with this number and after a while increasing it did not help much so we should do a profiling of it.

Reply:
For batch operations 20 threads are enough to fully load the DB. But now as the operations are sequential increasing the number of threads will help.

Reply:
ok, we will have to make some performance evaluation on a cluster with a large enough number of blocks.